dbms/programs/server/MySQLHandler.cpp

```diff
@@ -266,29 +267,78 @@ void MySQLHandler::comPing()
     packet_sender->sendPacket(OK_Packet(0x0, client_capability_flags, 0, 0, 0), true);
 }
 
-void MySQLHandler::comQuery(ReadBuffer & payload)
-{
-    bool with_output = false;
-    std::function<void(const String &)> set_content_type = [&with_output](const String &) -> void {
-        with_output = true;
-    };
+void MySQLHandler::comQuery(ReadBuffer &payload) {
+    std::string query = std::string(payload.position(), payload.buffer().end());
+
+    // This is a workaround in order to support adding ClickHouse to MySQL using federated server.
+    // As Clickhouse doesn't support these statements, we just send OK packet in response.
+    if (
+        (0 == strncasecmp("SET NAMES", query.c_str(), 9))
+        ||
+        (0 == strncasecmp("SET character_set_results", query.c_str(), 25))
+        ||
+        (0 == strncasecmp("SET FOREIGN_KEY_CHECKS", query.c_str(), 22))
+        ||
+        (0 == strncasecmp("SET AUTOCOMMIT", query.c_str(), 14))
+        ||
+        (0 == strncasecmp("SET SESSION TRANSACTION ISOLATION LEVEL", query.c_str(), 39))
+    ) 
```

Comment:
Should be moved to a static function isFederatedServerSetupCommand, to keep comQuery function short.

Reply:
done